src/components/IdentitiesSwitch.js

```diff
@@ -0,0 +1,310 @@
+					textStyle={fontStyles.t_regular}
+					style={styles.i_arrowStyle}
+				/>
+				<ButtonIcon
+					title="Privacy Policy"
+					onPress={() => closeModalAndNavigate('PrivacyPolicy')}
+					iconBgStyle={styles.i_arrowBg}
+					iconType="antdesign"
+					iconName="arrowright"
+					iconSize={18}
+					textStyle={fontStyles.t_regular}
+					style={styles.i_arrowStyle}
+				/>
+			</>
+		);
+	};
+
+	const renderNonSelectedIdentity = ({ item, index }) => {
+		const identity = item;
+		const title = identity.name || `identity_${index.toString()}`;
+
+		return (
+			<ButtonIcon
+				dropdown={false}
+				renderDropdownElement={() => renderIdentityOptions(identity)}
+				title={title}
+				onPress={() =>
+					onIdentitySelectedAndNavigate(identity, 'AccountNetworkChooser')
+				}
+				iconType="antdesign"
+				iconName="user"
+				iconSize={24}
+				style={styles.indentedButton}
+				textStyle={fontStyles.h2}
+			/>
+		);
+	};
+
+	const renderIdentities = () => {
+		// if no identity or the only one we have is the selected one
+
+		if (!identities.length || (identities.length === 1 && currentIdentity))
+			return;
+
+		const identitiesToShow = currentIdentity
+			? identities.filter(
+					identity => identity.encryptedSeed !== currentIdentity.encryptedSeed
+			  )
+			: identities;
+
+		return (
+			<>
+				<ScrollView style={{ maxHeight: 180 }}>
+					<FlatList
+						data={identitiesToShow}
+						renderItem={renderNonSelectedIdentity}
+						keyExtractor={item => item.encryptedSeed}
+						style={{ paddingVertical: identities.length > 5 ? 8 : 0 }}
+					/>
+				</ScrollView>
+				{identities.length > 5 && (
+					<Separator
+						shadow={true}
+						style={{ backgroundColor: 'transparent', marginTop: 0 }}
+						shadowStyle={{ opacity: 0.9 }}
+					/>
+				)}
+			</>
+		);
+	};
+
+	return (
+		<View>
+			<ButtonIcon
+				onPress={() => setVisible(!visible)}
+				iconName="user"
+				iconType="antdesign"
+				iconBgStyle={{ backgroundColor: 'transparent' }}
+				testID={testIDs.IdentitiesSwitch.toggleButton}
+			/>
+
+			<Modal
+				animationType="fade"
+				visible={visible}
+				transparent={true}
+				onRequestClose={() => setVisible(false)}
+			>
+				<TouchableWithoutFeedback
+					style={{ flex: 1 }}
+					onPressIn={() => setVisible(false)}
+				>
+					<View
+						testID={testIDs.IdentitiesSwitch.modal}
+						style={styles.container}
+						onPress={() => setVisible(false)}
+					>
+						<View style={styles.card}>
+							{renderCurrentIdentityCard()}
+							{renderIdentities()}
+							{accounts.getAccounts().size > 0 && (
+								<>
+									<ButtonIcon
+										title="Legacy Accounts"
+										onPress={() => closeModalAndNavigate('LegacyAccountList')}
+										iconName="solution1"
+										iconType="antdesign"
+										iconSize={24}
+										textStyle={fontStyles.t_big}
+										style={styles.indentedButton}
+									/>
+									<Separator />
+								</>
+							)}
+
+							{__DEV__ && (
+								<ButtonIcon
```

Comment:
Why for DEV only? is that an experimental feature for now?

Reply:
Thanks a lot! the dev only should be the `add legacy account` button but not this one